### PR TITLE
Add support for publishing pacts via patch request

### PR DESCRIPTION
**Pull request by ThePletch**
Adds a configuration setting for the pact publishing task that allows pacts to be published via a `PATCH` request, to support new functionality that merges pacts rather than overwriting them.

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,10 @@
+c
```
Comment (bethesque) [inline .byebug_history]: Accidental commit!

Comment (ThePletch) [inline .byebug_history]: Whoops.

**Comment by ThePletch** [comment]
Ready for re-review.

File: lib/pact_broker/client/pacts.rb [hunk 2]
```diff
@@ -11,7 +11,13 @@ def publish options
         pact_string = options[:pact_json]
         consumer_contract = ::Pact::ConsumerContract.from_json pact_string
         url = save_consumer_contract_url consumer_contract, consumer_version
-        response = self.class.put(url, body: pact_string, headers: default_put_headers)
+
+        if @client_options[:overwrite] == :merge
```
Comment (ThePletch) [inline lib/pact_broker/client/pacts.rb]: should this also be changed to `:write` instead of `:overwrite`?

Comment (bethesque) [inline lib/pact_broker/client/pacts.rb]: Yeah, I think that makes more sense.

Comment (bethesque) [inline lib/pact_broker/client/pacts.rb]: Happy to merge this once overwrite is replaced with write

**Comment by bethesque** [comment]
Just needs a fix in one of the tests. https://travis-ci.org/bethesque/pact_broker-client/jobs/152399898

**Comment by bethesque** [comment]
Thanks @ThePletch!

**Comment by bethesque** [comment]
Released 1.1.0

**Pull request merged**